config/src/test/scala/com/typesafe/config/impl/ConfigDocumentTest.scala

```diff
@@ -227,6 +227,30 @@ class ConfigDocumentTest extends TestUtils {
     }
 
     @Test
+    def configDocumentRemoveMultiple {
+        val origText = "a { b: 42 }, a.b = 43, a { b: { c: 44 } }"
+        val configDoc = ConfigDocumentFactory.parseString(origText)
+        val removed = configDoc.removeValue("a.b")
+        assertEquals("a { }, a { }", removed.render())
+    }
+
+    @Test
+    def configDocumentRemoveOverridden {
+        val origText = "a { b: 42 }, a.b = 43, a { b: { c: 44 }, a : 57 }"
+        val configDoc = ConfigDocumentFactory.parseString(origText)
+        val removed = configDoc.removeValue("a.b")
+        assertEquals("a { }, a { a : 57 }", removed.render())
```

Comment:
I believe it is in the original (the second object contained at a is the map `{ b: {c: 44}, a: 57}`)